pymc3/distributions/continuous.py

```diff
@@ -228,9 +228,32 @@ def __init__(self, *args, **kwargs):
         super(Flat, self).__init__(defaults=('_default',), *args, **kwargs)
 
     def random(self, point=None, size=None):
+        """Raises ValueError as it is not possible to sample from Flat distribution
+
+        Parameters
+        ----------
+        point : dict, optional
+        size : int, optional
+
+        Raises
+        -------
+        ValueError
+        """
         raise ValueError('Cannot sample from Flat distribution')
 
     def logp(self, value):
+        """
+        Calculate log-probability of Flat distribution at specified value.
+
+        Parameters
+        ----------
+        value : numeric
+            Value for which log-probability is calculated.
```

Comment:
`Value` --> `numpy or theano array`?

Reply:
Hey @junpenglao I don't quite understand. Are you asking that the type callout in the docstring be changed from numeric to numpy or theano array

Reply:
I meant changing `numeric` to `numpy array or theano tensor`.

Reply:
Got it. Will do and thanks for the comment

Reply:
Doing some testing it looks like this method can take python ints or floats as well so I don't feel that numpy array or theano tensor is completely descriptive. However if multiple values are passed they have to be in the data types you mentioned.

What do you guys think of a docstring that looks like this?
```
value : numeric
              Value(s) for which log-probability is calculated. If the log probabilities
              for multiples are desired value must be of type numpy array or theano tensor
```
<img width="831" alt="screen shot 2018-06-12 at 6 20 48 am" src="https://user-images.githubusercontent.com/7213793/41293482-362fc842-6e0a-11e8-871e-79bf8b47522c.png">

Reply:
Perfect!

Reply:
@junpenglao and @twiecki 
I have one more question on wording. Would it be more correct to say "log probability **density value**" instead of "log probability"?

From everything I understand for continuous distributions this function returns the log of the probability density value at that point, which is (apparently) **not** probability or log probability of that value. Worded another way the probability of single value in a PDF is zero, but the y value could be non zero.

I don't mean to be pedantic but I was really confused how the log probability of P(X=0) for a unit normal was -0.91 until I realized this was not probability but the PDF value at that point.

Please let me know if this makes sense.
Everything past this point is documentation a short list of my references and examples.

And again thanks for the quick replies and letting me contribute

Calculation of log PDF for a unit normal distribution
https://www.itl.nist.gov/div898/handbook/eda/section3/eda3661.htm

Code example
```
import pymc3 as pm
import numpy as np
y = pm.Normal.dist(mu=0, sd=1)
y.logp(0).eval()
np.log(np.exp(0)/np.sqrt(2*np.pi))
```

Succinct MathExchange posts explaining how probability at a point 
https://math.stackexchange.com/questions/1720053/how-can-a-probability-density-function-pdf-be-greater-than-1
https://stats.stackexchange.com/questions/86487/what-is-the-meaning-of-the-density-of-a-distribution-at-a-point

Reply:
density is not correct for a discrete distribution. I think saying just log probability is fine.